modules/library/api/src/main/java/org/geotools/geometry/jts/ReferencedEnvelope.java

```diff
@@ -925,10 +935,36 @@ public static ReferencedEnvelope reference(org.opengis.geometry.Envelope env) {
             return (ReferencedEnvelope) env;
         }
     
-        if (env.getDimension() == 3) {
-            return new ReferencedEnvelope3D(env);
+        if (env.getDimension() >= 3) {
+            // emptiness test according to com.vividsolutions.jts.geom.Envelope
+            if(env.getMaximum(0) < env.getMinimum(0)) {
+                return new ReferencedEnvelope3D(env.getCoordinateReferenceSystem());
+            } else {
+                return new ReferencedEnvelope3D(env);
+            }
         }
     
+        // emptiness test according to com.vividsolutions.jts.geom.Envelope
+        if(env.getMaximum(0) < env.getMinimum(0))
+            return new ReferencedEnvelope(env.getCoordinateReferenceSystem());
+        
         return new ReferencedEnvelope(env);
     }
+    
```

Comment:
Yes, it is. The constructor is also used in some places, where raster coordinates are transformed to screen coordinates (I don't know if an empty rectangle resulting is really a possible case). I have not problem with removing this method (too much API makes a library more complex and leads towards misunderstandings, that result in workarounds/solutions, which do not use the preferred way of doing things)